federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -765,7 +766,16 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 		desiredIngress.Spec = *objSpecCopy
 		glog.V(4).Infof("Desired Ingress: %v", desiredIngress)
 
-		if !clusterIngressFound {
+		send, err := clusterselector.SendToCluster(cluster.Labels, desiredIngress.ObjectMeta.Annotations)
+		if err != nil {
+			glog.Errorf("Error processing ClusterSelector cluster: %s for Ingress map: %s error: %s", cluster.Name, key, err.Error())
+			return
+		} else if !send {
+			glog.V(5).Infof("Skipping cluster: %s for Ingress: %s reason: cluster selectors do not match: %-v %-v", cluster.Name, key, cluster.ObjectMeta.Labels, desiredIngress.ObjectMeta.Annotations[federationapi.FederationClusterSelectorAnnotation])
```

Comment:
This glog is premature (and inaccurate) if we end up deleting an existing ingress on L812.